halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/NotificationService.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 URUP.com, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.config.services.v1;
+
+import com.netflix.spinnaker.halyard.config.model.v1.node.*;
+
+import com.netflix.spinnaker.halyard.config.model.v1.node.Notifications;
+import com.netflix.spinnaker.halyard.config.model.v1.notifications.SlackNotification;
+
+import com.netflix.spinnaker.halyard.core.problem.v1.ProblemSet;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+
+@Component
+public class NotificationService {
+
+    @Autowired
+    private LookupService lookupService;
+
+    @Autowired
+    private DeploymentService deploymentService;
+
+    @Autowired
+    private ValidateService validateService;
+
+    public void setNotification(String deploymentName, Notification notification) {
+        DeploymentConfiguration deploymentConfiguration = deploymentService.getDeploymentConfiguration(deploymentName);
+        Notifications notifications = deploymentConfiguration.getNotifications();
+        switch (notification.notificationType()) {
+            case SLACK:
+                notifications.setSlack((SlackNotification) notification);
+                break;
+            default:
+                throw new IllegalArgumentException("Unknonwn notification type " + notification.notificationType());
+        }
+    }
+
+    public void setEnabled(String deploymentName, String notificationName, boolean enabled) {
+        Notification notification = getNotification(deploymentName, notificationName);
```

Comment:
I think you're missing a `getNotification`. Should be straightforward - look at the `SecurityService` for an example